src/client/opamClient.ml

```diff
@@ -635,6 +635,57 @@ let init_checks ?(hard_fail_exn=true) init_config =
   if hard_fail && hard_fail_exn then OpamStd.Sys.exit_because `Configuration_error
   else not (soft_fail || hard_fail)
 
+let git_for_windows_check =
+  if Sys.win32 then
+    fun cygbin ->
+      let gitcmd = OpamSystem.resolve_command "git" in
```

Comment:
To deal with the common edge case where both Cygwin (or MSYS2) git and Git for Windows git are installed, I've been doing PATH modifications in DkML so that Git for Windows always is in front of the PATH (before `/usr/bin`). That may be overkill here but the problem remains what to do when they are both installed.

Currently this PR will search the PATH and only check the first `git` ... which will almost always be Cygwin/MSYS2 `/usr/bin/git`.

Not sure how, but could `OpamSystem.resolve_command` be adjusted to find the correct `git`?

Reply:
FYI: The PATH hack for shells (`PATH="${PATH/:\/usr\/bin:\/bin:/:}"`) seen in https://learn.microsoft.com/en-us/vcpkg/users/platforms/mingw#how-to-avoid-mixing-different-installations seems like a common way to fix these sorts of Cygwin/MSYS2 problems. So my suggestion would be to have a `OpamSystem.resolve_command ~safe:()` mode which does the PATH hack in OCaml to find if there is a Git for Windows. And if that fails, then do a normal `OpamSystem.resolve_command` if needed.